### DISCUSSION TITLE
prPriority of oldest packages first

### DISCUSSION THREAD

**Discussion by WesleyYue** [renovatebot/renovate#25458] [Suggest an Idea]
### Tell us more.

I want to setup renovatebot such that it'll prioritize opening PRs for the most stale packages in my repo.

**Comment by HonkingGoose** [comment]
By default, Renovate has a `prHouryLimit` of 2, and a `prConcurrentLimit` of 10. This means that Renovate will create two PRs each time it runs, until it has opened 10 PRs. I think Renovate only prioritizes security/vulnerability type updates.

You can always get a update PR for any package by selecting the update in the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/).

I recommend you read [our upgrade best practices](https://docs.renovatebot.com/upgrade-best-practices/) to learn how to deal with a repository with old packages.

I'll let the maintainers answer your feature request to prioritize stale packages.

**Reply by WesleyYue** [reply]
Thanks, I'm familiar with everything you mentioned but it doesn't fit our needs. We have a slower pr open rate/concurrency that works best for us but leads to renovate continuously upgrading the same packages that have very high release cadences and blocking the rest.

**Reply by HonkingGoose** [reply]
> Thanks, I'm familiar with everything you mentioned but it doesn't fit our needs. 

Oh okay. 😞 Let's try again! 😄 

> We have a slower pr open rate/concurrency that works best for us but leads to renovate continuously upgrading the same packages that have very high release cadences and blocking the rest.

I recommend updating _more_ often, because its faster and easier overall. If you want to keep the slow cadence, see if any of these things look good:

- Set `prConcurrentLimit` to `0`, so Renovate will eventually create PRs for all your dependencies. Renovate will limit the PRs per hour to the `prHourlyLimit`, which defaults to `2`. But this will cost you more build/test minutes because Renovate rebases each PR when it gets stale. If you're paying for test/build minutes, be careful!
- Read [Renovate's Noise Reduction docs](https://docs.renovatebot.com/noise-reduction/) to see if you can reduce the overall number of PRs.
- [Set a `prPriority` for your important packages](https://docs.renovatebot.com/configuration-options/#prpriority), to make sure you're getting those updates before others.
- Remember that you can always get any update you want from the Dependency Dashboard.

> ## prPriority
> 
> Sometimes Renovate needs to rate limit its creation of PRs, e.g. hourly or concurrent PR limits.
> By default, Renovate sorts/prioritizes based on the update type, going from smallest update to biggest update.
> Renovate creates update PRs in this order:
> 
> 1. `pinDigest`
> 1. `pin`
> 1. `digest`
> 1. `patch`
> 1. `minor`
> 1. `major`
> 
> If you have dependencies that are more or less important than others then you can use the `prPriority` field for PR sorting.
> The default value is `0`, so setting a negative value will make dependencies sort last, while higher values sort first.
> 
> Here's an example of how you would define PR priority so that `devDependencies` are raised last and `react` is raised first:
> 
> ```json
> {
>   "packageRules": [
>     {
>       "matchDepTypes": ["devDependencies"],
>       "prPriority": -1
>     },
>     {
>       "matchPackageNames": ["react"],
>       "prPriority": 5
>     }
>   ]
> }
> ```
